src/resolver/nodeModuleLookup.ts

```diff
@@ -112,104 +112,82 @@ export function findTargetFolder(props: IResolverProps, parsed: IModuleParsed):
     if (fileExists(path.join(attempted, "package.json"))) {
-      return attempted;
+      return realpathSync(attempted);
     }
   }
 }
 export interface INodeModuleLookup {
   error?: string;
+  // TODO: not used?
   isEntry?: boolean;
   meta?: IPackageMeta;
   targetAbsPath?: string;
+  // TODO: not used?
   targetExtension?: string;
+  // TODO: not used?
   targetFuseBoxPath?: string;
 }
 
 export function nodeModuleLookup(props: IResolverProps, parsed: IModuleParsed): INodeModuleLookup {
   let folder: string;
+  const { name: moduleName, target } = parsed;
 
-  folder = findTargetFolder(props, parsed);
+  // Resolve the module name to a folder
+  folder = findTargetFolder(props, moduleName);
   if (!folder) {
-    return { error: `Cannot resolve "${parsed.name}"` };
+    return { error: `Cannot resolve "${moduleName}"` };
   }
 
-  const result: INodeModuleLookup = {};
-  const pkg: IPackageMeta = {
-    name: parsed.name,
-    packageRoot: folder,
-  };
-  result.meta = pkg;
-
   const packageJSONFile = path.join(folder, 'package.json');
   if (!fileExists(packageJSONFile)) {
-    return { error: `Failed to find package.json in ${folder} when resolving module ${parsed.name}` };
+    return { error: `Failed to find package.json in ${folder} when resolving module ${moduleName}` };
   }
   const json = JSON.parse(readFile(packageJSONFile));
-  pkg.version = json.version || '0.0.0';
-  pkg.browser = json.browser;
-  pkg.packageJSONLocation = packageJSONFile;
-  pkg.packageRoot = folder;
-  if (json['fuse-box']) {
-    pkg.fusebox = json['fuse-box'];
-  }
+  // Prepare the package metadata structure and copy a bunch of stuff into it
+  const pkg: IPackageMeta = {
+    name: moduleName,
+    packageRoot: folder,
+    version: json.version || '0.0.0',
+    browser: json.browser,
+    packageJSONLocation: packageJSONFile,
+    fusebox: json['fuse-box'] || undefined,
+  };
 
   const isBrowser = props.buildTarget === 'browser';
-  // extract entry point
-
-  // extract target if required
-  if (parsed.target) {
-    const parsedLookup = fileLookup({ fileDir: folder, target: parsed.target });
-    if (!parsedLookup) {
-      return { error: `Failed to resolve ${props.target} in ${parsed.name}` };
-    }
-
-    result.targetAbsPath = parsedLookup.absPath;
-
-    if (isBrowser && json.browser && typeof json.browser === 'object') {
-      const override = handleBrowserField(pkg, parsedLookup.absPath);
-      if (override) {
-        result.targetAbsPath = override;
-        parsedLookup.customIndex = true;
-      }
-    }
-
-    result.isEntry = false;
-    result.targetFuseBoxPath = makeFuseBoxPath(folder, result.targetAbsPath);
-  } else {
-    const entryFile = getFolderEntryPointFromPackageJSON({ isBrowserBuild: isBrowser, json: json });
 
-    const entryLookup = fileLookup({ fileDir: folder, target: entryFile });
+  const isEntry = !target;
+  const targetResolver = props.tsTargetResolver || fileLookup;
```

Comment:
so if references are set it's going to use typescript for resolving the files? Could it potentially disrupt the lookup defined by FUseBox, like aliases e.t.c

Reply:
Aliases are processed first, and the output of processing aliases is what is sent to `nodeModuleLookup()`.  So aliasing can interfere with Typescript, but not the other way around.

The TypeScript resolver just does its own aliasing and passes the result onto the default resolver.  Its own aliasing is automatically aliases output files to input files.  So the order here means that if you alias to an output file, it will be aliased again (by typescript) to an input file.  I believe this is the desired behavior.

For some examples:
  1. If I alias `"file.js"` to `"dist/file.js"` then the typescript resolver will further change it from `"dist/file.js"` to `"src/file.ts"` and load that.
  2. If I alias `"file.ts"` to `"src/file.ts"` then the typescript resolver will just pass that through unchanged because it isn't an output.

The only potential issue, I think, is if someone tries to alias an input file to another input file, but still tries to import the output file. But I don't think this is a problem, because it already wouldn't work today, and it still is the expected behavior, I think.  The scenario looks like this:
  1. If I alias `"src/file.ts"` to `"src/other-file.ts"` and then I `import "src/file.ts"` this will work, but if I `import "dist/file.js"` then typescript will change it to `"src/file.ts"` but it won't go *back through* aliasing, so what will be loaded is `"src/file.ts"` not `"src/other-file.ts"`.

Reply:
Sounds awesome man, one thing I'd ask you to do to help me review it.
Could you add a comprehensive example into the `playground` folder? I will check out the branch and see the work in action.

You're becoming the man behind the fuse-box module resolution ;-) I trust your skills, but I want a playground example mainly because I want to understand how it works!

Reply:
Yeah, I should be able to do that.  I have my own playground project I have been working with, so I will polish that up a bit and see if it will do the trick.